java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/utils/ClientAuthenticationUtils.java

```diff
@@ -101,6 +106,117 @@ private static CredentialCallOption getAuthenticate(final FlightClient client,
     return factory.getCredentialCallOption();
   }
 
+  private static KeyStore getKeyStoreInstance(String instance)
+      throws KeyStoreException, CertificateException, IOException, NoSuchAlgorithmException {
+    KeyStore keyStore = KeyStore.getInstance(instance);
+    keyStore.load(null, null);
+
+    return keyStore;
+  }
+
+  private static String getOperatingSystem() {
+    return System.getProperty("os.name");
+  }
+
+  /**
+   * Check if the operating system running the software is Windows.
+   *
+   * @return whether is the windows system.
+   */
+  public static boolean isWindows() {
+    return getOperatingSystem().contains("Windows");
+  }
+
+  /**
+   * Check if the operating system running the software is Mac.
+   *
+   * @return whether is the mac system.
+   */
+  public static boolean isMac() {
+    return getOperatingSystem().contains("Mac");
+  }
+
+  /**
+   * Check if the operating system running the software is Linux.
+   *
+   * @return whether is the linux system.
+   */
+  public static boolean isLinux() {
+    return getOperatingSystem().contains("Linux");
+  }
+
+  /**
+   * Gets the trusted certificates input stream from the JVM.
+   *
+   * @return the trusted certificates as an {@link InputStream}.
+   * @throws IOException                if it fails reading the file.
+   * @throws KeyStoreException          if a key store could not be loaded.
+   * @throws CertificateException       if a certificate could not be found.
+   */
+  public static InputStream getCertificateInputStreamFromJVM()
+      throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
+    Path path = Paths.get(System.getProperty("java.home"), "lib", "security", "cacerts");
+    InputStream fileInputStream = Files.newInputStream(path);
+    KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+    String password = "changeit";
```

Comment:
We should use the supplied password as the user may have changed it (they are supposed to).

Reply:
so if the user is using the useSystemTrustStore in linux, he also needs to pass the password? Since he needs it to access the truststore in the jvm

Reply:
Yes

Reply:
fixed